redash/models.py

```diff
@@ -1,238 +1,133 @@
-
-        return changes
-
-    def save(self, *args, **kwargs):
-        changed_by = kwargs.pop('changed_by', None)
-        pk_value = self._get_pk_value()
-        created = kwargs.get('force_insert', False) or not bool(pk_value)
-
-        if created and changed_by is None:
-            changed_by = self.user
-
-        ret = super(ChangeTrackingMixin, self).save(*args, **kwargs)
-
-        if changed_by:
-            Change.log_change(changed_by, self)
-            self._clean_values = {}
-
-        return ret
-
-    def update_instance(self, **kwargs):
-        changed_by = kwargs.pop('changed_by', None)
-        ret = super(ChangeTrackingMixin, self).update_instance(**kwargs)
-        if changed_by:
-            Change.log_change(changed_by, self)
-        return ret
-
-    def _field_names(self):
-        return [f.name for f in self._meta.get_fields()]
-
+                changes[k] = {'previous': v, 'current': getattr(self, k)}
+        session.add(Change(object_type=self.__class__.__tablename__,
+                           object_id=self.id,
+                           object_version=self.version,
+                           user_id=changed_by.id,
+                           change=changes))
+        session.add(self)
 
 
 class ConflictDetectedError(Exception):
     pass
 
-
-class BaseVersionedModel(BaseModel):
-    version = peewee.IntegerField(default=1)
-
-    def save(self, *args, **kwargs):
-        pk_value = self._get_pk_value()
-        created = kwargs.get('force_insert', False) or not bool(pk_value)
-
-        if created:
-            # Since this is an `INSERT`, just call regular save method.
-            return super(BaseVersionedModel, self).save()
-
-        # Update any data that has changed and bump the version counter.
-        self.pre_save(False)
-
-        field_data = dict(self._data)
-        current_version = field_data.pop('version', 0)
-        field_data = self._prune_fields(field_data, self.dirty_fields)
-
-        # if not field_data:
-        #     raise ValueError('No changes have been made.')
-
-        ModelClass = type(self)
-        field_data['version'] = ModelClass.version + 1  # Atomic increment
-
-        query = ModelClass.update(**field_data).where(
-            (ModelClass.version == current_version) &
-            (ModelClass.id == self.id))
-
-        nrows = query.execute()
-        if nrows == 0:
-            # It looks like another process has updated the version number.
-            raise ConflictDetectedError()  # Raise exception? Return False?
-        else:
-            self.version += 1  # Update in-memory version number.
-            self._dirty.clear()
-            self.post_save(False)
-            return nrows
-
-
 class BelongsToOrgMixin(object):
     @classmethod
     def get_by_id_and_org(cls, object_id, org):
-        return cls.get(cls.id == object_id, cls.org == org)
+        return cls.query.filter(cls.id == object_id, cls.org == org).first()
```

Comment:
Will this query ever return more than one result? If not, should probably use `one()` or `one_or_none()`